packages/astro/test/fixtures/middleware-no-user-middleware/integration-middleware-url.js

```diff
@@ -0,0 +1,13 @@
+import { defineMiddleware } from 'astro:middleware';
+
+export const onRequest = defineMiddleware((context, next) => {
+	if (context.url.pathname === '/url') {
+		return new Response(JSON.stringify({ post: 'works' }), {
+			headers: {
+				'content-type': 'application/json',
+			},
+		});
```

Comment:
Not requiring changing, but this can also be shorten as 
```suggestion
		return Response.json({ post: 'works' });
```